requirements.txt

```diff
@@ -6,3 +6,5 @@ watchdog==0.8.3
 # Use sckoo@stanford.edu's patched argcomplete, until pull request has been accepted
 # https://github.com/kislyuk/argcomplete/pull/118
 git+https://github.com/kashizui/argcomplete.git@allow-overridden-subparsersaction
+# Use agd@stanford.edu's fork of hash_ring with extra features
+git+https://github.com/andreweduffy/hash_ring.git@master
```

Comment:
There are no good consistent hashing libraries for Python :/ (see https://github.com/Doist/hash_ring/issues/11)

I added the ability and dynamically add and remove nodes in the hash ring, which doesn't exist in the original version. If you'd like I can move the changes over into CodaLab, but the original repo that I forked from doesn't have any license, which apparently means that "default copyright rules apply" i.e. you can't use their code directly https://help.github.com/articles/open-source-licensing/#what-happens-if-i-dont-choose-a-license. I doubt that's what the author intended, but just thought you should be aware.

I can give the ownership of the repo to CodaLab, and then we can maintain our own fork if you think that'd be better.